### PR TITLE
Add FAQ from Aspire discussion

### LINKED ISSUE #332: Add to the FAQ
There is some content we can add to the FAQ, based on this discussion here:

https://github.com/dotnet/aspire/discussions/1038

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Fixes #332 - Preview: [Frequently asked questions about .NET Aspire](https://review.learn.microsoft.com/en-us/dotnet/aspire/reference/aspire-faq?branch=pr-en-us-333)

### CODE REVIEW

File: docs/reference/aspire-faq.yml [hunk 1]
```diff
@@ -3,14 +3,43 @@ metadata:
...
+
+      - question: |
+          Why would I use .NET Aspire service discovery when Docker Compose has it built in and works with Kubernetes?
+        answer: |
+          .NET Aspire service discovery APIs are an abstraction that works with various providers (like Kubernetes and Consul). One of the big advantages is that it works locally, and is backed by .NET's `IConfiguration` abstraction. This means you can implement service discovery across your compute fabric in a way that doesn't result in code changes. What happens to your application code if you have multiple kube clusters or services on app services, azure functions, ECS etc? Do I have to fundamentally change it to make it work locally, in a single cluster or across multiple? Well, no you don't. That's the benefit of the abstraction.
```
Comment (gewarren) [inline docs/reference/aspire-faq.yml:26]:
```suggestion
          .NET Aspire service discovery APIs are an abstraction that works with various providers (like Kubernetes and Consul). One of the big advantages is that it works locally and is backed by .NET's `IConfiguration` abstraction. This means you can implement service discovery across your compute fabric in a way that doesn't result in code changes. If you have multiple Kubernetes clusters or services on Azure App Service or Azure Functions, you don't have to fundamentally change your application code to make it work locally, either in a single cluster or across multiple clusters. That's the benefit of the abstraction.
```

File: docs/reference/aspire-faq.yml [hunk 2]
```diff
@@ -3,14 +3,43 @@ metadata:
...
+
+      - question: |
+           OpenTelemetry is something that can already be used in .NET. Why would I use .NET Aspire?
+        answer: |
+           We didn't invent this, .NET Aspire takes a big bet on .NET's integration with OpenTelemetry. Our dashboard is a standard OTLP server that visualizes various telemetry data. Leaning on these open standards makes it very easy to build these things without breaking compatibility with the broader ecosystem.
```
Comment (gewarren) [inline docs/reference/aspire-faq.yml:31]:
```suggestion
           .NET Aspire takes a big bet on .NET's integration with OpenTelemetry. The .NET Aspire dashboard is a standard OTLP server that visualizes various telemetry data. Leaning on these open standards makes it easy to build these things without breaking compatibility with the broader ecosystem.
```

Comment (gewarren) [reply docs/reference/aspire-faq.yml:31]: What does "these things" refer to?

File: docs/reference/aspire-faq.yml [hunk 3]
```diff
@@ -3,14 +3,43 @@ metadata:
...
+
+      - question: |
+          Observability such as Grafana, Jaeger, and Prometheus work with .NET, why bother with .NET Aspire?
```
Comment (gewarren) [inline docs/reference/aspire-faq.yml:34]:
```suggestion
          Observability such as Grafana, Jaeger, and Prometheus work with .NET. Why bother with .NET Aspire?
```

File: docs/reference/aspire-faq.yml [hunk 4]
```diff
@@ -3,14 +3,43 @@ metadata:
...
+        answer: |
+          .NET Aspire isn't a replacement for these sorts of things, but rather a complementary technology. .NET Aspire is a set of libraries and tools that make it easy to build applications that are observable. For more information, see the [Metrics example in the .NET Aspire sample repository](https://github.com/dotnet/aspire-samples/tree/main/samples/Metrics) that shows Grafana and Prometheus.
```
Comment (gewarren) [inline docs/reference/aspire-faq.yml:36]:
```suggestion
          .NET Aspire isn't a replacement for these tools, but rather a complementary technology. .NET Aspire is a set of libraries and tools that make it easy to build applications that are observable. For more information, see the [Metrics example in the .NET Aspire sample repository](https://github.com/dotnet/aspire-samples/tree/main/samples/Metrics) that shows Grafana and Prometheus.
```

File: docs/reference/aspire-faq.yml [hunk 5]
```diff
@@ -3,14 +3,43 @@ metadata:
...
+
+      - question: |
+          Why is there a need for a yet another framework to do what's already being done very well by everyone else? 
+        answer: |
+          .NET Aspire isn't a framework. Perhaps the most controversial part of it's the `DistributedApplication` APIs that are a way to build up the orchestration model in any .NET based language. we can build a better developer experience. While everything is possible today, it's not easy. The entire cloud native ecosystem is built around tying various lego pieces of CNCF software together to build a stack (unix philosophy). You can think of aspire as trying to do the same thing, using what we've come to understand about the cloud native space and picking some opinions (in ways that use the same building blocks). I think one novel thing about how we're building various pieces of the stack is that we're really not making it a "walled garden". As people play with it more, they'll realize how composable and extensible it is.
```
Comment (gewarren) [inline docs/reference/aspire-faq.yml:41]:
```suggestion
          .NET Aspire isn't a framework. Perhaps the most controversial part of it are the `DistributedApplication` APIs that you can use to build up the orchestration model in any .NET-based language. While everything is possible today, it's not easy. Using the Unix philosophy, the entire cloud-native ecosystem is built around tying various pieces of CNCF software together to build a stack. .NET Aspire tries to do the same thing using learnings from the cloud-native space and picks some opinions (in ways that use the same building blocks). One novel thing about how .NET Aspire builds various pieces of the stack is that it's not a "walled garden". As people play with it more, they realize how composable and extensible it is.
```

Comment (gewarren) [reply docs/reference/aspire-faq.yml:41]: Not sure if "walled garden" is a universal term or if it would translate well.

**Review by gewarren — CHANGES_REQUESTED** [review]

File: docs/reference/aspire-faq.yml [hunk 6]
```diff
@@ -3,14 +3,43 @@ metadata:
...
 
 sections:
   - name: Frequently asked questions
     questions:
+      - question: |
+          Why use .NET Aspire for orchestration when I can use Docker Compose?
+        answer: |
+          Docker compose is excellent but is unproductive when all you want to do is run several projects or executables. Docker compose requires developers to build container images, and run inside of containers. That's a barrier when I just want to run my frontend, backend, workers and a database. You don't need to learn anything else beyond what you already know.
+
+          Just imagine trying to call your backend API from your frontend API (using a simple `http://backend` url) and having to under docker compose in order to enable that? It's too much.
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply docs/reference/aspire-faq.yml:19]: I didn't think that FAQ had to be in docs voice, I thought it was more product voice?

File: docs/reference/aspire-faq.yml (hunk 5 above)
Comment (IEvangelist) [reply docs/reference/aspire-faq.yml]: Good catch, I believe you're right.

File: docs/reference/aspire-faq.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/reference/aspire-faq.yml]: I don't think so.

File: docs/reference/aspire-faq.yml [hunk 7]
```diff
@@ -3,14 +3,43 @@ metadata:
...
+
+      - question: |
+          Why is there a need for a yet another framework to do what's already being done very well by everyone else? 
+        answer: |
+          .NET Aspire isn't a framework. Perhaps the most controversial part of it are the `DistributedApplication` APIs that you can use to build up the orchestration model in any .NET-based language. While everything is possible today, it's not easy. Using the Unix philosophy, the entire cloud-native ecosystem is built around tying various pieces of CNCF software together to build a stack. .NET Aspire tries to do the same thing using learnings from the cloud-native space and picks some opinions (in ways that use the same building blocks). One novel thing about how .NET Aspire builds various pieces of the stack is that it's not a "walled garden". As people play with it more, they realize how composable and extensible it is.
```
Comment (IEvangelist) [inline docs/reference/aspire-faq.yml:41]:
```suggestion
          .NET Aspire isn't a framework. Perhaps the most controversial part of it are the `DistributedApplication` APIs that you can use to build up the orchestration model in any .NET-based language. While everything is possible today, it's not easy. Using the Unix philosophy, the entire cloud-native ecosystem is built around tying various pieces of CNCF software together to build a stack. .NET Aspire tries to do the same thing using learnings from the cloud-native space and picks some opinions (in ways that use the same building blocks). One novel thing about how .NET Aspire builds various pieces of the stack is that it doesn't restrict the access or compatibility of other applications, frameworks, or services. As people play with it more, they realize how composable and extensible it is.
```

File: docs/reference/aspire-faq.yml [hunk 8]
```diff
@@ -3,14 +3,43 @@ metadata:
...
 summary: |
   This article lists frequently asked questions about .NET Aspire. For a more comprehensive overview, see [.NET Aspire overview](../get-started/aspire-overview.md).
 
 sections:
   - name: Frequently asked questions
     questions:
+      - question: |
+          Why use .NET Aspire for orchestration when I can use Docker Compose?
+        answer: |
+          Docker compose is excellent but is unproductive when all you want to do is run several projects or executables. Docker compose requires developers to build container images and to run apps inside of containers. That's a barrier when you just want to run your front end, back end, workers, and a database. With .NET Aspire, you don't need to learn anything else beyond what you already know.
```
Comment (gewarren) [inline docs/reference/aspire-faq.yml:17]:
```suggestion
          Docker Compose is excellent but is unproductive when all you want to do is run several projects or executables. Docker Compose requires developers to build container images and to run apps inside of containers. That's a barrier when you just want to run your front end, back end, workers, and a database. With .NET Aspire, you don't need to learn anything beyond what you already know.
```

File: docs/reference/aspire-faq.yml [hunk 9]
```diff
@@ -3,14 +3,43 @@ metadata:
...
+
+          Imagine trying to call your back-end API from your front-end API (using a simple `http://backend` URL) and having to use Docker Compose to enable that&mdash;it's too much.
```
Comment (gewarren) [inline docs/reference/aspire-faq.yml:19]: Personally I'd just leave this line out.

File: docs/reference/aspire-faq.yml [hunk 10]
```diff
@@ -3,14 +3,43 @@ metadata:
...
+
+          Configuration through declarative code is better than YAML. Docker Compose gets complex once you attempt to do any form of abstraction or composition (for example, see the old [eshopOnContainers app](https://github.com/dotnet-architecture/eShopOnContainers/tree/dev/src)). In addition, there are environment variable replacements (and includes), no types or IntelliSense, and it's hard to reason about what exactly is running. Debugging is also difficult. .NET Aspire produces a better experience that's easy to get started and scales up to an orchestrator like Compose using a real programming language.
```
Comment (gewarren) [inline docs/reference/aspire-faq.yml:21]:
```suggestion
          Configuration through declarative code is better than through YAML. Docker Compose gets complex once you attempt to do any form of abstraction or composition (for example, see the old [eshopOnContainers app](https://github.com/dotnet-architecture/eShopOnContainers/tree/dev/src)). In addition, there are environment variable replacements (and includes) and no types or IntelliSense, and it's hard to reason about what exactly is running. Debugging is also difficult. .NET Aspire produces a better experience that's easy to get started and scales up to an orchestrator like Compose using a real programming language.
```

**Review by gewarren — APPROVED** [review]

File: docs/reference/aspire-faq.yml (hunk 9 above)
Comment (IEvangelist) [inline docs/reference/aspire-faq.yml:19]:
```suggestion
```

**Pull request merged**